### PR TITLE
Align event delivery with cloud events specification

### PR DESCRIPTION
**Pull request by montaro**
See also #4388

### CODE REVIEW

File: tests/event-bus/e2e-tester/e2e-tester.go [hunk 1]
```diff
@@ -36,10 +36,10 @@ const (
...
 	retries = 20
 
-	sourceIDHeader         = "Ce-Source-Id"
-	eventTypeHeader        = "Ce-Event-Type"
-	eventTypeVersionHeader = "Ce-Event-Type-Version"
-	customHeader           = "Ce-X-Custom-Header"
+	sourceIDHeader         = "ce-source"
+	eventTypeHeader        = "ce-type"
+	eventTypeVersionHeader = "ce-eventtypeversion"
+	customHeader           = "ce-xcustomheader"
```
Comment (sayanh) [inline tests/event-bus/e2e-tester/e2e-tester.go]: Following are the missing headers:
`ce-specversion`
`ce-id`
`ce-time`
`content-type`
IMO we should test as per the contract as mentioned in [examples](https://github.com/kyma-project/examples/pull/129/commits/5c2ca1263022687f7273903c2fb2e1dc59f09a11)

File: docs/event-bus/03-03-service-programming-model.md [hunk 2]
```diff
@@ -23,10 +23,10 @@ The following HTTP Headers provide information about the Event metadata.
 
 |Header| Description|
 |------|--------|
-| **ce-event-id** | Business Event's ID delivered to the microservice. |
-| **ce-event-time** | Business Event's time delivered to the microservice. |
-| **ce-event-type** | Business Event's type delivered to the microservice. |
-| **ce-event-type-version** | Business Event's version delivered to the microservice. |
+| **ce-id** | Business Event's ID delivered to the microservice. |
```
Comment (sayanh) [inline docs/event-bus/03-03-service-programming-model.md]: The headers [here](https://github.com/kyma-project/kyma/pull/4585/files#r296583732) should be added as well.

File: tests/event-bus/e2e-tester/e2e-tester.go (hunk 1 above)
Comment (montaro) [reply tests/event-bus/e2e-tester/e2e-tester.go]: Thanks for the catch! 
Done.

File: docs/event-bus/03-03-service-programming-model.md (hunk 2 above)
Comment (montaro) [reply docs/event-bus/03-03-service-programming-model.md]: Done!

**Pull request merged**